hedera-node/src/main/java/com/hedera/services/txns/token/TokenWipeTransitionLogic.java

```diff
@@ -46,34 +51,53 @@
  */
 public class TokenWipeTransitionLogic implements TransitionLogic {
 	private static final Logger log = LogManager.getLogger(TokenWipeTransitionLogic.class);
-
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
-
-	private final TokenStore store;
 	private final TransactionContext txnCtx;
+	private final TypedTokenStore tokenStore;
+	private final AccountStore accountStore;
+	private final OptionValidator validator;
+	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
 
 	public TokenWipeTransitionLogic(
-			TokenStore store,
-			TransactionContext txnCtx
+			final OptionValidator validator,
+			final TypedTokenStore tokenStore,
+			final AccountStore accountStore,
+			final TransactionContext txnCtx
 	) {
-		this.store = store;
 		this.txnCtx = txnCtx;
+		this.tokenStore = tokenStore;
+		this.accountStore = accountStore;
+		this.validator = validator;
 	}
 
 	@Override
 	public void doStateTransition() {
-		try {
-			var op = txnCtx.accessor().getTxn().getTokenWipe();
-			var outcome = store.wipe(op.getAccount(), store.resolve(op.getToken()), op.getAmount(),false);
-			var id = store.resolve(op.getToken());
-			txnCtx.setStatus((outcome == OK) ? SUCCESS : outcome);
-			if(outcome == OK) {
-				txnCtx.setNewTotalSupply(store.get(id).totalSupply());
-			}
-		} catch (Exception e) {
-			log.warn("Unhandled error while processing :: {}!", txnCtx.accessor().getSignedTxnWrapper(), e);
-			txnCtx.setStatus(FAIL_INVALID);
+		/* --- Translate from gRPC types --- */
+		var op = txnCtx.accessor().getTxn().getTokenWipe();
+		final var grpcTokenId = op.getToken();
+		final var grpcAccountId = op.getAccount();
+		final var targetTokenId = new Id(grpcTokenId.getShardNum(), grpcTokenId.getRealmNum(), grpcTokenId.getTokenNum());
+		final var targetAccountId = new Id(grpcAccountId.getShardNum(), grpcAccountId.getRealmNum(), grpcAccountId.getAccountNum());
+
+		/* --- Load the model objects --- */
+		final var token = tokenStore.loadToken(targetTokenId);
+		final var account = accountStore.loadAccount(targetAccountId);
+		final var accountRel = tokenStore.loadTokenRelationship(token, account);
+
+		/* --- Instantiate change trackers --- */
+		final var ownershipTracker = new OwnershipTracker();
+
+		/* --- Do the business logic --- */
+		if (token.getType().equals(TokenType.FUNGIBLE_COMMON)) {
+			token.wipe(accountRel, op.getAmount());
+		} else {
+			token.wipe(ownershipTracker, accountRel, op.getSerialNumbersList());
```

Comment:
In order to resolve the bug mention in the other comment, I think that the best solution might be to do:
- `tokenStore.loadUniqueTokens(token, op.getSerialNumbersList());`
Under the hood, this would add the `List<UniqueTokens>` to the `Token` model. In the `Token` model we can add `List<UniqueTokens> loadedUniqueTokens`

Reply:
I was just checking out how to obtain those tokens :) Thanks, I'll try it